### PR TITLE
Create an inter-region travel model file format

### PR DESCRIPTION
**Pull request by jonathanvdc**
This PR's pretty boring. I created some data structures, an accompanying file format and a couple of unit tests. Additionally, I noticed and refactored a common XML file parsing pattern. That's all, I guess.

### CODE REVIEW

**Review by cekefun — COMMENTED** [review]
The std::vector<AirRoute> in the airport have no description. Neither do the variables in RegionTravel

**Comment by jonathanvdc** [comment]
AFAICT, the `std::vector` in `struct Airport` _does_ have a description:

```cpp
/// Gets the list of all outgoing routes that start at this airport.
std::vector<AirRoute> routes;
```

Also, those (admittedly undocumented) fields are private. So who'd read the documentation?

**Edit:** My reasoning for not documenting those private fields was that doing so anyway would be redundant. The values they represent are already documented by their getters. Simply copy-pasting those doc-comments might seem like a good idea in the short term, but later the docs might diverge, which would be confusing.

```cpp
/// Gets the region id for the region this data structure represents.
RegionId GetRegionId() const { return region_id; }

/// Gets the path of the population file for this region.
std::string GetRegionPopulationPath() const { return region_population_path; }

/// Gets the fraction of people in the region who travel by plane on any given day.
double GetTravelFraction() const { return travel_fraction; }

/// Gets a list of all airports.
const std::vector<AirportRef>& GetAllAirports() const { return *all_airports; }

/// Gets a list of airports in the current region.
const std::vector<AirportRef>& GetLocalAirports() const { return local_airports; }

/// Gets the set of region ids for regions that have at least one air
/// route which targets a local airport.
const std::unordered_set<RegionId>& GetRegionsWithIncomingRoutes() const
{
	return regions_with_incoming_routes;
}
```

File: src/main/cpp/util/InstallDirs.cpp [hunk 1]
```diff
@@ -167,5 +170,21 @@ path InstallDirs::GetRootDir()
 	return g_root_dir;
 }
 
+/// Reads the XML file at the given path, relative to some other path.
```
Comment (lynn) [inline src/main/cpp/util/InstallDirs.cpp]: Maybe “some other path” → “some anchor path”, to make it instantly obvious which path goes where when looking at the parameter names?

Comment (lynn) [reply src/main/cpp/util/InstallDirs.cpp]: Actually, I suppose we should remove this bit of documentation and only keep the copy in the header file.

**Review by lynn — APPROVED** [review]
This looks great! These tests are very thorough :+1:

**Pull request merged**